### PR TITLE
Use border-box for input, textarea to revert radio buttons to normal appearance

### PR DESCRIPTION
**Pull request by danielFangstrom**
## Description
Recent changes to the styling of meta boxes caused certain input fields to have a strange appearance, such as radio buttons having their selection dot off-centre. This commit changes the box-sizing back to border-box, which is in line with standard Wordpress appearance. 

## How Has This Been Tested?
Manually checked the radio buttons to make sure that the selection is centred, and made sure other fields still look the same.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style.
- [x] My code has proper inline documentation.

### CODE REVIEW

**Comment by aduth** [comment]
Related: #5581

**Comment by aduth** [comment]
Can we be very explicit about what elements we're expecting should have which `box-sizing`, and importantly _why_ we're making that assumption. It's my understanding this is for backwards compatibility with existing meta box styling. Where is that styling defined? And for what elements are `box-sizing` exemptions made?

**Review by atimmer — CHANGES_REQUESTED** [review]
@aduth Well in WordPress there wasn't a style that said either `box-sizing: border-box` or `box-sizing: content-box`. This meant that the browser styles decided what the `box-sizing` would be of an element. Gutenberg puts `box-sizing` to `border-box`, so to match the old behavior in the meta box section `box-sizing` needs to be set to `content-box`. Except for a few elements, such as `input` and `textarea`.

@danielFangstrom I think `button` should probably also be added to this list, according to this issue: https://github.com/necolas/normalize.css/issues/73.

Because meta boxes are explicitly a way to offer backward compatibility I think we should do our best to mimic the environment the old meta boxes were in.

**Comment by aduth** [comment]
Would I be correct in assuming the invalid sizes from https://github.com/Yoast/gutenberg/issues/45 impact Webkit browsers only? If so, would it be enough to set `box-sizing: initial`, since the [unsupported browser](https://developer.mozilla.org/en-US/docs/Web/CSS/initial#Browser_compatibility) (IE11) is not affected anyways?

```css
.edit-post-meta-boxes-area {
	* {
		box-sizing: content-box;
		box-sizing: initial;
	}
}
```

**Comment by danielFangstrom** [comment]
@aduth `box-sizing: initial` does not appear to help, as the default value for `box-sizing` is `content-box`. However, certain elements get overwritten with `border-box` later on by stylesheets and to replicate the appearance I added `border-box` to these elements, which are `input`, `textarea`, and `.button`.  I might have overlooked additional ones.

**Comment by aduth** [comment]
Where are these elements being overridden? If we're assuming to apply a top-level `box-sizing: border-box` from the Gutenberg wrapper, but exempt meta boxes area, we shouldn't have other `box-sizing` being applied that we risk conflicting with.

**Comment by danielFangstrom** [comment]
It seems `input` and `textarea` are both set to `border-box` via forms.css, while `.button` is set to `border-box` in buttons.css (which I now see is more specific than both `.gutenberg *` and `.edit-post-meta-boxes-area *`, so my addition in this commit is not really necessary). I haven't found any other styles being applied in between forms.css and Gutenberg, at least not for metaboxes .
Since Gutenberg overrides forms.css for these elements it looks like we should have 
```
input, texture { 
       box-sizing: border-box
}
```
in the metaboxes-area/style.scss in order to have the appearance restored.

**Review by aduth — CHANGES_REQUESTED** [review]
>while .button is set to border-box in buttons.css

If you're referring to these styles, then we also need `.button-primary` and `.button-secondary`:

https://github.com/WordPress/WordPress/blob/9a11487e22f75a890b7875e835d7ee055f9fe779/wp-includes/css/buttons.css#L43-L60

Since this discussion has been rather time consuming to track down the _why_ of the change, can we leave a code comment for the next person to have some references?

File: edit-post/components/meta-boxes/meta-boxes-area/style.scss [hunk 1]
```diff
@@ -2,10 +2,17 @@
 .edit-post-meta-boxes-area {
 	position: relative;
 
+    /** The wordpress default for most meta-box elements is content-box. Some elements, such as textarea and input, are
```
Comment (aduth) [inline edit-post/components/meta-boxes/meta-boxes-area/style.scss]: Spaces should be tab.

I recommend installing an [EditorConfig plugin](http://editorconfig.org/#download) for your preferred editor, which will automatically adjust to using tabs for the project specifically.

Comment (aduth) [reply edit-post/components/meta-boxes/meta-boxes-area/style.scss]: Updated in 52dab6e

**Review by aduth — APPROVED** [review]

**Comment by aduth** [comment]
@atimmer Any remaining thoughts?

**Review by atimmer — APPROVED** [review]
No, I think it is now where it needs to be.

**Pull request merged**